imblearn/over_sampling/_adasyn.py

```diff
@@ -131,7 +130,9 @@ def _fit_resample(self, X, y):
                 )
             ratio_nn /= np.sum(ratio_nn)
             n_samples_generate = np.rint(ratio_nn * n_samples).astype(int)
-            if not np.sum(n_samples_generate):
+            # rounding may cause new amount for n_samples
+            n_samples = np.sum(n_samples_generate)
```

Comment:
Original non-vectorized implementation used this approach to make n-samples rather than truncating `np.sum(n_samples_generate)` down to `n_samples`